io_scene_niftools/kf_export.py

```diff
@@ -0,0 +1,103 @@
+#      or promote products derived from this software without specific
+#      prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# ***** END LICENSE BLOCK *****
+
+import os
+import bpy
+
+import pyffi.spells.nif.fix
+
+from io_scene_niftools.file_io.kf import KFFile
+from io_scene_niftools.modules.nif_export import armature
+from io_scene_niftools.modules.nif_export.animation.transform import TransformAnimation
+from io_scene_niftools.nif_common import NifCommon
+from io_scene_niftools.utils import math
+from io_scene_niftools.utils.singleton import NifOp, NifData
+from io_scene_niftools.utils.logging import NifLog, NifError
+from io_scene_niftools.modules.nif_export import scene
+
+
+class KfExport(NifCommon):
+
+    def __init__(self, operator, context):
+        NifCommon.__init__(self, operator, context)
+
+        # Helper systems
+        self.transform_anim = TransformAnimation()
+
+    def execute(self):
+        """Main import function."""
+
+        NifLog.info(f"Exporting {NifOp.props.filepath}")
+
+        # extract directory, base name, extension
+        directory = os.path.dirname(NifOp.props.filepath)
+        filebase, fileext = os.path.splitext(os.path.basename(NifOp.props.filepath))
+
+        prefix = "x" if bpy.context.scene.niftools_scene.game in ('MORROWIND',) else ""
+        self.version, data = scene.get_version_data()
+        NifData.init(data)
+
+        b_armature = math.get_armature()
+        # some scenes may not have an armature, so nothing to do here
+        if b_armature:
+            math.set_bone_orientation(b_armature.data.niftools.axis_forward, b_armature.data.niftools.axis_up)
+
+        NifLog.info("Creating keyframe tree")
+        kf_root = self.transform_anim.export_kf_root(b_armature)
+
+        # write kf (and xkf if asked)
+        ext = ".kf"
+        NifLog.info(f"Writing {prefix}{ext} file")
+
+        data.roots = [kf_root]
+        data.neosteam = (bpy.context.scene.niftools_scene.game == 'NEOSTEAM')
+
+        # scale correction for the skeleton
+        if bpy.context.scene.niftools_scene.game in ('SKYRIM'):
+            toaster = pyffi.spells.nif.NifToaster()
+            toaster.scale = round(1 / NifOp.props.scale_correction)
```

Comment:
pull out `1 / NifOp.props.scale_correction` to local variable `scale`
as also used on line 96

Reply:
Should probably be moved out to a function anyway so nif and kf export use the same

Reply:
Would be a good improvement alright

Reply:
Done!